### PR TITLE
Fix warnings

### PR DESCRIPTION
**Pull request by usbalbin**
Fix clippy warnings

### CODE REVIEW

**Comment by no111u3** [comment]
@usbalbin this review partially outdated, please update if you have some ability.

**Comment by usbalbin** [comment]
```
warning: method `unlock_options` is never used
  --> src/flash.rs:62:8
   |
61 | impl<'a, const SECTOR_SZ_KB: u32> FlashWriter<'a, SECTOR_SZ_KB> {
   | --------------------------------------------------------------- method in this implementation
62 |     fn unlock_options(&mut self) -> Result<()> {
   |        ^^^^^^^^^^^^^^
```

@amcelroy , Is this function supposed to be public or is it just left as a "good to have" for features to be added later?

File: src/pwm.rs [hunk 1]
```diff
@@ -1080,7 +1080,7 @@ fn calculate_deadtime(base_freq: Hertz, deadtime: NanoSecond) -> (u8, u8) {
     let deadtime_ticks = deadtime_ticks as u64 * 429497;
     let deadtime_ticks = (deadtime_ticks >> 32) as u32;
 
-    let deadtime_ticks = deadtime_ticks as u32;
+    let deadtime_ticks = deadtime_ticks;
```
Comment (no111u3) [inline src/pwm.rs:1083]: Why we didn't remove it?

File: src/rcc/mod.rs [hunk 2]
```diff
@@ -328,6 +328,7 @@ impl AHB1 {
     fn rstr(rcc: &RccRB) -> &rcc::AHB1RSTR {
         &rcc.ahb1rstr
     }
+    #[allow(unused)]
```
Comment (no111u3) [inline src/rcc/mod.rs:331]: Doesn't need since I merge one of my PR's

File: src/rcc/mod.rs [hunk 3]
```diff
@@ -346,6 +347,7 @@ impl AHB2 {
     fn rstr(rcc: &RccRB) -> &rcc::AHB2RSTR {
         &rcc.ahb2rstr
     }
+    #[allow(unused)]
```
Comment (no111u3) [inline src/rcc/mod.rs:350]: Same.

File: src/rcc/mod.rs [hunk 4]
```diff
@@ -364,6 +366,7 @@ impl AHB3 {
     fn rstr(rcc: &RccRB) -> &rcc::AHB3RSTR {
         &rcc.ahb3rstr
     }
+    #[allow(unused)]
```
Comment (no111u3) [inline src/rcc/mod.rs:369]: Same

File: src/rcc/mod.rs [hunk 5]
```diff
@@ -382,6 +385,7 @@ impl APB1_1 {
     fn rstr(rcc: &RccRB) -> &rcc::APB1RSTR1 {
         &rcc.apb1rstr1
     }
+    #[allow(unused)]
```
Comment (no111u3) [inline src/rcc/mod.rs:388]: Same

File: src/rcc/mod.rs [hunk 6]
```diff
@@ -400,6 +404,7 @@ impl APB1_2 {
     fn rstr(rcc: &RccRB) -> &rcc::APB1RSTR2 {
         &rcc.apb1rstr2
     }
+    #[allow(unused)]
```
Comment (no111u3) [inline src/rcc/mod.rs:407]: Same

File: src/rcc/mod.rs [hunk 7]
```diff
@@ -418,6 +423,7 @@ impl APB2 {
     fn rstr(rcc: &RccRB) -> &rcc::APB2RSTR {
         &rcc.apb2rstr
     }
+    #[allow(unused)]
```
Comment (no111u3) [inline src/rcc/mod.rs:426]: Same

**Review by no111u3 — CHANGES_REQUESTED** [review]
@usbalbin please mark comments as fixed if you fix it, it make clear that exactly changed.

File: Cargo.toml [hunk 8]
```diff
@@ -101,3 +101,7 @@ debug = false
 codegen-units = 1
 incremental = false
 lto = true
+
+[[example]]
+name = "flash_with_rtic"
+required-features = ["stm32g474"]
```
Comment (usbalbin) [inline Cargo.toml:107]: Not sure if this is how something like this is best done...

**Review by no111u3 — APPROVED** [review]

**Pull request merged**